lesson26/src/js/login.js

```diff
@@ -1,12 +1,55 @@
 import * as validation from "./validation";
 import { togglePassword } from "./module/toggle-password";
+import { Chance } from "chance";
+const chance = new Chance();
 
 const submitButton  = document.querySelector('[data-id="submit_button"]');
+const userNameInput = document.getElementById('username');
+const passwordInput = document.getElementById('current-password');
+const togglePasswordButton = document.getElementById('js-toggle-password');
 
 submitButton.addEventListener("click", (e) => {
     e.preventDefault();
+    login();
 });
 
+const isRegisteredUser = (inputValues) => {
+    const userData = {
+        name:"Nakagawa",
+        email: "nakagawa@sample.com",
+        password: "N302aoe3"
+    }
+
+    return ( inputValues.userName === userData.name || userData.email ) && inputValues.password === userData.password;
+}
+
+const checkRegisteredUser = () => {
+    const inputValues = {
+        userName : userNameInput.value,
+        password : passwordInput.value
+    }
+
+    return new Promise((resolve,reject) => {
+        if( isRegisteredUser(inputValues) ){
+            resolve({ token: chance.apple_token(), ok: true , code: 200 })
+        }else{
+            reject({ ok: false, code: 401 })
+        }
+    }) 
+}
+
+const login = async () => {
+    try{
+        const response = await checkRegisteredUser();
+        if(!response) return;
+        localStorage.setItem("token",JSON.stringify(response.token));
+
+        window.location.href = "./index.html";
+    }catch{
+        window.location.href = "./401.html";
+    }
+}
```

Comment:
On success, the token is embedded in local storage and the contents page is displayed; on failure, the user is redirected to page 401.